application/src/main/java/org/togetherjava/tjbot/commands/moderation/attachment/BlacklistedAttachmentListener.java

```diff
@@ -0,0 +1,110 @@
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.moderation.ModAuditLogWriter;
+
+import java.awt.*;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+/**
+ * Reacts to blacklisted attachments being posted, upon which they are deleted.
+ */
+public class BlacklistedAttachmentListener extends MessageReceiverAdapter {
+    private final ModAuditLogWriter modAuditLogWriter;
+    private final List<String> blacklistedFileExtensions;
+
+    /**
+     * Creates a AttachmentListener to receive all message sent in any channel.
+     *
+     * @param config to find the blacklisted media attachments
+     * @param modAuditLogWriter to inform the mods about the suspicious attachment
+     */
+    public BlacklistedAttachmentListener(Config config, ModAuditLogWriter modAuditLogWriter) {
+        super(Pattern.compile(".*"));
+        this.modAuditLogWriter = modAuditLogWriter;
+        this.blacklistedFileExtensions = config.getBlacklistedFileExtensions();
+    }
+
+    @Override
+    public void onMessageReceived(MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+        if (doesMessageContainBlacklistedContent(event.getMessage())) {
+            handleBadMessage(event.getMessage());
+        }
+    }
+
+    private void handleBadMessage(Message message) {
+        message.delete().flatMap(any -> dmUser(message)).queue(any -> warnMods(message));
+    }
+
+    private RestAction<Message> dmUser(Message message) {
+        Message dmMessage = createDmMessage(message);
+        return message.getAuthor()
+            .openPrivateChannel()
+            .flatMap(privateChannel -> privateChannel.sendMessage(dmMessage));
+    }
+
+    private Message createDmMessage(Message originalMessage) {
+        String contentRaw = originalMessage.getContentRaw();
+        String blacklistedAttachments =
+                String.join(", ", getBlacklistedAttachmentsFromMessage(originalMessage));
+        String dmMessageContent =
+                """
+                        Hey there, you posted a message containing a blacklisted file attachment: %s.
+                        We had to delete your message for security reasons.
+
+                        Feel free to repost your message without, or with a different file instead. Sorry for any inconvenience caused by this 🙇️
+                        """
+                    .formatted(blacklistedAttachments);
+        // No embed needed if there was no message from the user
+        if (contentRaw.isEmpty()) {
+            return new MessageBuilder(dmMessageContent).build();
+        }
+        return createBaseResponse(contentRaw, dmMessageContent);
```

Comment:
a bit hard to read on a small screen (github CR) - can u add some newlines to separate the sections? thanks